### ISSUE TITLE
The index of <pad>、<UNK> in the vocabulary about adaptive softmax

### ISSUE THREAD

**Issue by cswangjiawei** [pytorch/fairseq#868]
In the paper "Language Modeling with Gated Convolutional Networks", adaptive softmax is adopted, which requires that the words with the highest frequency should have the smallest index. In the implementation of this paper,'<S>'and'</S>' will be added at the beginning and end of each sequence, and zero-pad the begining of the sequence with k-1 elements. So, in the vocabulary, should the index of <pad>, <S>, </S>, <UNK> be the beginning or the end of the vocabulary?
